update/update-from-2.16-to-2.17.php

```diff
@@ -0,0 +1,170 @@
+	if (!$link) {
+		return false;
+	}
+
+	$result = pg_query($link, "select version from config");
+
+	if ($result != NULL) {
+		$version = pg_fetch_array($result);
+		if (strcmp($version["version"], "2.16") == 0) {
+			return true;
+		}
+	}
+	return false;
+}
+// run upgrade scripts
+
+require_once(PHPREPORT_ROOT . 'config/config.php');
+
+
+if (!check_db_version(DB_HOST,DB_PORT,DB_NAME,DB_USER,DB_PASSWORD)) {
+	print ("Wrong database version. " .
+		"Make sure DB is on 2.16 version before running this upgrade.\n");
+	exit();
+}
+
+
+
+require_once(PHPREPORT_ROOT . 'config/config.php');
+include_once(PHPREPORT_ROOT . '/util/DBPostgres.php');
+
+if (!check_db_version(DB_HOST,DB_PORT,DB_NAME,DB_USER,DB_PASSWORD)) {
+	print ("Wrong database version. " .
+		"Make sure DB is on 2.1 version before running this upgrade.\n");
+	exit();
+}
+
+$success = true;
+foreach ($sqlFiles as $file) {
+	if (!parse_psql_dump($file,DB_HOST,DB_PORT,DB_NAME,DB_USER,DB_PASSWORD)) {
+		$success = false;
+		break;
+	}
+}
+// finish, print message
+
+if ($success) {
+
+	$link = pg_connect( "host=" . DB_HOST . " port=" . DB_PORT . " user=" . DB_USER . " dbname=" . DB_NAME . " password=" . DB_PASSWORD );
+
+	if ( !$link ) {
+		error_log( "ERROR: Could not connect to DB" );
+	}
+
+	$result = pg_query( $link, $query = "SELECT customerid, projectid FROM requests" );
+	if ( $result == null ) {
+		error_log( "ERROR: Could not run query: $query" );
+	}
+
+	$customerAssignedToProject = array();
+
+	while ( $row = pg_fetch_array( $result, null, PGSQL_ASSOC ) ) {
+		$customerAssignedToProject[$row['projectid']][] = $row['customerid'];
+	}
+
+	foreach ( $customerAssignedToProject as $projectId => $value ) {
+		$customerAssociated = $value[0];
+		if ( count( $customerAssignedToProject[$projectId] ) > 1 ) {
+			$taskCustomerFreq = array();
+			foreach ( $value as $customerId => $dummy ) {
+				// We have more than one customers for these projects, so lets check
+				// in tasks table
+				$result = pg_query( $link, $query = "select customerid, count(*) as freq from task where projectid=$projectId group by customerid;" );
+				$rows = pg_fetch_all( $result );
+
+				foreach ( $rows as $row ) {
+					// There are chances that null shows up, lets skip it
+					if ( $row['customerid'] != null ) {
+						$taskCustomerFreq[$projectId][$row['customerid']] = $row['freq'];
+					}
+				}
+			}
+
+			//We sort by value to get the highest number of customer association
+			foreach ( $taskCustomerFreq as $projectFreq ) {
+				arsort( $projectFreq );
+				$taskCustomerFreq[$projectId] = $projectFreq;
+			}
+
+			if ( isset( $taskCustomerFreq[$projectId] ) ) {
+				$customerAssociated = key( $taskCustomerFreq[$projectId] );
+			}
+		}
+		if ($customerAssociated) {
+			$update_query = "UPDATE project SET customerid= $customerAssociated where id= $projectId";
+			print( $update_query . "\n" );
+			if ( !pg_query( $link, $update_query ) ) {
+				error_log( "ERROR: query failed: $update_query" );
+			}
```

Comment:
You are mixing code styles in this file, be careful with `if ( $customerAssociated )` and `if ($customerAssociated)` and the like. I prefer the latter, which is more widely used in the code base.